### PR TITLE
Php unit testing

### PR DESCRIPTION
**Pull request by hassan7867**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -24,8 +25,9 @@
             "lib/"
         ],
         "psr-4": {
-            "Jsys\\Types\\": "lib/",
-            "Jsys\\Types\\Http\\": "lib/http/"
+            "Jsys\\Types\\": "",
```
Comment (jsyspk) [inline composer.json]: remove empty mapping

File: lib/strings/non-zero-length-string.php [hunk 2]
```diff
@@ -0,0 +1,25 @@
+<?php
+declare(strict_types=1);
+use jsys\types\strings\NonZeroLengthStringInterface;
+
+class NonZeroLengthString implements NonZeroLengthStringInterface
+{
+    private $string;
+
+    public function __construct(string $string)
```
Comment (jsyspk) [inline lib/strings/non-zero-length-string.php]: take TextInterface as param

File: lib/strings/text.php [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+{
+    private $str;
+    public function __construct(string $str)
+    {
+        {
+            $this->str = $str;
+        }
+    }
+
+    public function isEmpty(string $str)
```
Comment (jsyspk) [inline lib/strings/text.php]: ` public function isEmpty(): bool`

File: lib/strings/text.php [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    {
+        if (!empty($str))
+        {
+            throw new \InvalidArgumentException("Non empty '$str' is not Allowed", 1021);
+
+        }
+        return true;
+    }
+
+    public function length(string $str)
```
Comment (jsyspk) [inline lib/strings/text.php]: ` public function length(): int`

File: lib/strings/text.php [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+    public function value(): string
+    {
+        return $this->str;
+    }
+
+    /**
+     * @return string
+     */
+    public function getStr(): string
```
Comment (jsyspk) [inline lib/strings/text.php]: change to __toString()

File: lib/strings/white-space-string.php [hunk 6]
```diff
@@ -0,0 +1,26 @@
+<?php
+declare(strict_types=1);
+
+use jsys\types\strings\WhiteSpaceStringInterface;
+
+class WhiteSpaceString implements WhiteSpaceStringInterface
+{
+    private $string;
+
+    public function __construct(string  $string)
```
Comment (jsyspk) [inline lib/strings/white-space-string.php]: TextInterface as param

File: lib/strings/zero-length-string.php [hunk 7]
```diff
@@ -0,0 +1,26 @@
+<?php
+declare(strict_types=1);
+use jsys\types\strings\ZeroLengthStringInterface;
+
+class ZeroLengthString implements ZeroLengthStringInterface
+{
+    private $string;
+
+    public function __construct(string $string)
```
Comment (jsyspk) [inline lib/strings/zero-length-string.php]: TextInterface as param

**Review by jsyspk — CHANGES_REQUESTED** [review]

File: lib/strings/text.php [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+{
+    private $str;
+    public function __construct(string $str)
+    {
+        {
+            $this->str = $str;
+        }
+    }
+
+    public function isEmpty(string $str):bool
```
Comment (zenexjp) [inline lib/strings/text.php]: what ???

File: lib/strings/text.php [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+        }
+        return true;
+    }
+
+    public function length(string $str):int
+    {
+        return mb_strlen($str);
+    }
+    public function hasSpaces(string $str)
+    {
```
Comment (zenexjp) [inline lib/strings/text.php:40]: nope
is Space and has space are two different  functions

File: tests/unit/strings/is-empty-string-test.php [hunk 10]
```diff
@@ -0,0 +1,38 @@
+<?php
+declare(strict_types=1);
+
+namespace Jsys\strings\Test;
+use jsys\types\strings\Text;
+use PHPUnit\Framework\TestCase;
+
+
+class IsEmptyTest extends TestCase
+{
```
Comment (zenexjp) [inline tests/unit/strings/is-empty-string-test.php:10]: makes no sense

File: tests/unit/strings/non-white-space-string-test.php [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+        {
+        $testString = new Text(" ");
+        $this->expectExceptionCode(1002);
+        $this->expectExceptionMessage("Given string ".$testString->value()." contain non white space.");
+        $nonWhiteString = new NonWhiteSpaceString($testString);
+    }
+
+    public function test_can_initialize_with_non_white_space_string()
+    {
+        $testString = new Text("");
```
Comment (zenexjp) [inline tests/unit/strings/non-white-space-string-test.php:28]: are you sure this test is OK?

File: tests/unit/strings/non-zero-length-string-test.php [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+    public function test_must_not_initialize_with_empty_string_parameter()
+    {
+        $testString = new Text('');
+        $this->expectExceptionCode(1001);
+        $this->expectExceptionMessage("Given string ".$testString->value()." is empty.");
+        $EmptyStr = new NonZeroLengthString($testString);
+    }
+
+    public function test_can_initialize_with_non_empty_string_parameter()
+    {
```
Comment (zenexjp) [inline tests/unit/strings/non-zero-length-string-test.php:27]: give some more params

File: tests/unit/strings/string-length-test.php [hunk 13]
```diff
@@ -0,0 +1,30 @@
+<?php
+declare(strict_types=1);
+
+namespace Jsys\strings\Test;
+use jsys\types\strings\Text;
+use PHPUnit\Framework\TestCase;
+
+
+class StringLengthTest extends TestCase
+{
```
Comment (zenexjp) [inline tests/unit/strings/string-length-test.php:10]: what is this test?

**Review by zenexjp — CHANGES_REQUESTED** [review]

File: tests/unit/strings/is-empty-string-test.php [hunk 14]
```diff
@@ -0,0 +1,38 @@
+<?php
+declare(strict_types=1);
+
+namespace Jsys\strings\Test;
```
Comment (khizerabbas) [inline tests/unit/strings/is-empty-string-test.php:4]: namespace should be either all lowercase or all titlecase

File: tests/unit/strings/is-empty-string-test.php (hunk 10 above)
Comment (hassan7867) [reply tests/unit/strings/is-empty-string-test.php:10]: sorry?

File: lib/strings/text.php [hunk 15]
```diff
@@ -0,0 +1,53 @@
+<?php
+declare(strict_types=1);
+
+
+namespace jsys\types\strings;
+
+class Text implements TextInterface
+{
+    private $str;
```
Comment (khizerabbas) [inline lib/strings/text.php:9]: Please write complete variable names i.e $string

File: tests/unit/strings/non-white-space-string-test.php (hunk 11 above)
Comment (hassan7867) [reply tests/unit/strings/non-white-space-string-test.php:28]: Yes test is working

File: tests/unit/strings/string-length-test.php (hunk 13 above)
Comment (hassan7867) [reply tests/unit/strings/string-length-test.php:10]: checking the length of the string

File: lib/strings/white-space-string.php [hunk 16]
```diff
@@ -0,0 +1,26 @@
...
+use jsys\types\strings\WhiteSpaceStringInterface;
+use jsys\types\strings\TextInterface;
+class WhiteSpaceString implements WhiteSpaceStringInterface
+{
+    private $string;
+
+    public function __construct(TextInterface  $string)
+    {
+        if(!ctype_space($string->value())) {
+            throw new \InvalidArgumentException("Given string ".$string->value()." doesnot contain white space.", 1001);
```
Comment (khizerabbas) [inline lib/strings/white-space-string.php:13]: Grammar mistake: Use does't or does not

File: tests/unit/strings/has-spaces-test.php [hunk 17]
```diff
@@ -0,0 +1,38 @@
+<?php
+declare(strict_types=1);
+namespace Jsys\strings\Test;
```
Comment (khizerabbas) [inline tests/unit/strings/has-spaces-test.php]: namespace should be either all lowercase or all titlecase

File: lib/strings/text.php [hunk 18]
```diff
@@ -0,0 +1,53 @@
...
+    public function __construct(string $str)
+    {
+        {
```
Comment (jsyspk) [inline lib/strings/text.php]: why double braces ?

File: lib/strings/text.php (hunk 8 above)
Comment (jsyspk) [reply lib/strings/text.php]: what are you trying to do here?
can you fix it ?

File: lib/strings/text.php (hunk 9 above)
Comment (jsyspk) [reply lib/strings/text.php:40]: can you fix it?

**Review by jsyspk — CHANGES_REQUESTED** [review]

**Pull request merged**